src/unix_fileid.rs

```diff
@@ -0,0 +1,59 @@
+
+use crate::Error;
+use crate::sys::FileDesc;
+
+use std::{sync::{Arc, Mutex, Condvar}, collections::{HashMap, hash_map::Entry}};
+use once_cell::sync::Lazy;
+
+pub type FileId = (u64, u64);
```

Comment:
I believe we're not required to have specifically `u64`s for `FileId`, it would be ok to use libc's types directly and we would not need to risk a bad bit-cast.

```suggestion
pub type FileId = (libc::dev_t, libc::ino_t);
```